app/views/links/_details.html.erb

```diff
@@ -46,22 +46,40 @@
   </h2>
 
   <% if link.post_thumbnail_url && link.post_url %>
-    <figure>
+    <figure data-controller="embiggen-toggle">
       <a href="<%= link.post_url %>" target="_blank" class="background-sample">
         <% case device %>
         <% when :desktop %>
-          <ion-icon role="presentation" name="laptop-outline" class="background-sample__icon--desktop"></ion-icon>
+          <ion-icon role="presentation" name="laptop-outline" class="background-sample__icon--desktop hide-when-embiggened"></ion-icon>
         <% when :mobile %>
-          <ion-icon role="presentation" name="phone-portrait-outline" class="background-sample__icon--mobile"></ion-icon>
+          <ion-icon role="presentation" name="phone-portrait-outline" class="background-sample__icon--mobile hide-when-embiggened"></ion-icon>
         <% end %>
 
-        <img src="<%= link.post_thumbnail_url %>" class="background-sample__image--<%= device %>"/>
+        <img src="<%= link.post_thumbnail_url %>" class="embiggen-me-daddy background-sample__image--<%= device %>"/>
       </a>
       <figcaption class="link--description">
+      <div class="embiggen-button-container"> 
+        <button type="button" data-action="click->embiggen-toggle#toggle">
+          <ion-icon name="expand" class="text-embiggen"></ion-icon>
+          <ion-icon name="contract" class="text-ensmallen"></ion-icon>
+          <span class="show-for-warren show-for-perverts text-embiggen">MAKE IT HUGE!</span>
```

Comment:
Normally you'd check the mascot/perv status within the markup itself, but this is rendered within a turbo frame, and AFAIK the codebase doesn't have any precedent set on how to handle that. So I went ahead and just did the stupid-simple thing, and turned the mascot state into classes that go onto the root html element.

Turns the potentially-complicated Turbo pain into some simple CSS. Doesn't scale, but for mascot-related things, this should be fine.

Reply:
I LOVE THIS, this is so cute

Reply:
And yeah, you'd have to render this in a partial, so you could send it in a turbo frame over the wire on a hook from the user model, which would get complicated. This is a smart solution!